sklearnex/_device_offload.py

```diff
@@ -179,9 +201,12 @@ def wrapper(self, *args, **kwargs):
         if len(data) == 0:
             usm_iface = None
         else:
-            usm_iface = getattr(data[0], '__sycl_usm_array_interface__', None)
+            usm_iface = getattr(data[0], "__sycl_usm_array_interface__", None)
         result = func(self, *args, **kwargs)
         if usm_iface is not None:
-            return _copy_to_usm(usm_iface['syclobj'], result)
+            result = _copy_to_usm(usm_iface["syclobj"], result)
+            if dpnp_available and isinstance(data[0], dpnp.ndarray):
+                result = dpnp.array(result, copy=False)
```

Comment:
This is and related dpnp import are only functional changes in this file, the rest is formatting.